src/core/unit.c

```diff
@@ -1522,6 +1523,34 @@ bool unit_shall_confirm_spawn(Unit *u) {
         return !unit_get_exec_context(u)->same_pgrp;
 }
 
+static bool unit_verify_deps(Unit *u) {
+        static const UnitDependency negative[] = {
+                UNIT_CONFLICTS,
+                UNIT_CONFLICTED_BY,
+        };
+
+        Unit *other;
+        unsigned i;
+        Iterator j;
+
+        assert(u);
+
+        /* Checks whether all strict dependencies of this unit hold. This checks only BindsTo=, Conflicts= and the
+         * reverse of the latter. We do not check Requires= or Requisite= here as they only should have an effect on
+         * the job processing, but do not have any effect afterwards. */
+
+        SET_FOREACH(other, u->dependencies[UNIT_BINDS_TO], j)
+                if (!UNIT_IS_ACTIVE_OR_RELOADING(unit_active_state(other)))
```

Comment:
I think we should log something here. Something like https://github.com/systemd/systemd/pull/4733#issuecomment-263001559
Note: I logged an incorrect message for `!UNIT_IS_INACTIVE_OR_FAILED(unit_active_state(other))` case